content/en/docs/concepts/workloads/pods/pod-lifecycle.md

```diff
@@ -460,14 +460,25 @@ An example flow:
       The containers in the Pod receive the TERM signal at different times and in an arbitrary
       order. If the order of shutdowns matters, consider using a `preStop` hook to synchronize.
       {{< /note >}}
-1. At the same time as the kubelet is starting graceful shutdown, the control plane removes that
-   shutting-down Pod from EndpointSlice (and Endpoints) objects where these represent
+1. At the same time as the kubelet is starting graceful shutdown, the control plane evaluates
+   whether a terminating Pod needs to be removed from any Endpoints of the
+   EndpointSlice object which represent
    a {{< glossary_tooltip term_id="service" text="Service" >}} with a configured
    {{< glossary_tooltip text="selector" term_id="selector" >}}.
    {{< glossary_tooltip text="ReplicaSets" term_id="replica-set" >}} and other workload resources
    no longer treat the shutting-down Pod as a valid, in-service replica. Pods that shut down slowly
-   cannot continue to serve traffic as load balancers (like the service proxy) remove the Pod from
-   the list of endpoints as soon as the termination grace period _begins_.
+   should not continue to serve regular traffic and should be terminating connection.
+   Some pods require the graceful termination - for example session draining and completion. Without the feature
+   gate `EndpointSliceTerminatingCondition` (in beta starting Kubernetes version `1.22`),
```

Comment:
```suggestion
   Some pods work best when you use graceful termination - for example, you might want that
   for graceful session draining and completion. If you don't have the the
   `EndpointSliceTerminatingCondition` feature gate enabled in your cluster, then
```

Reply:
This maybe a nit. However, do we want to use graceful twice in the first sentence? Session draining & completion, afaik, should be graceful?